File: client/src/Packets/DATAPacket.cpp
#include "../../include/Packets/DATAPacket.h"

DATAPacket::DATAPacket(short size, short block, char *bytes):data(bytes),blockNum(block),packetSize(size) {
    this->opCode = 3;
}
DATAPacket::DATAPacket(const DATAPacket &dATAPacket):packetSize(dATAPacket.packetSize),
                                                     blockNum(dATAPacket.blockNum),
                                                     data(dATAPacket.data){}

DATAPacket::DATAPacket(short opCode, short size, short block, char* bytes):data(bytes),
                                                                           blockNum(block),packetSize(size) {
    this->opCode = opCode;
}

short DATAPacket::getPacketSize() {
    return packetSize;
}


short DATAPacket::getBlockNum() {
    return blockNum;
}



char* DATAPacket::getData() {
    return data;
}
DATAPacket::~DATAPacket() {}

File: server/src/main/java/bgu/spl171/net/impl/Packets/DATAPacket.java
package bgu.spl171.net.impl.Packets;

public class DATAPacket extends Packet
{
	short packetsize;
	short block;
	byte[] data;
	
	public DATAPacket(short op, short packetsize, short block, byte[] data) 
	{
		super(op);
		this.packetsize = packetsize;
		this.block = block;
		this.data = data;
	}
	
	public void setBlock(short b)
	{
		block = b;
	}
	
	public short getBlock()
	{
		return block;
	}
	
	public short getSize()
	{
		return packetsize;
	}
	
	public byte[] getData()
	{
		return data;
	}
}
